build.gradle

```diff
@@ -246,34 +251,68 @@ tasks.cleanEclipseJdt {
   }
 }
 
-if (project.hasProperty('privateToolkitUsername') && project.hasProperty('privateToolkitPassword')) {
+artifacts {
+  archives javadocJar, sourcesJar
+}
+
+signing {
+  required { gradle.taskGraph.hasTask("uploadArchives") }
+  sign configurations.archives
+}
+
+if (project.hasProperty('ossrhUsername') && project.hasProperty('ossrhPassword')) {
   uploadArchives {
     repositories {
       mavenDeployer {
-        snapshotRepository(url: toolkitSnapshotRepo) {
-          authentication(userName: privateToolkitUsername, password: privateToolkitPassword)
+        beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
+
+        repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
+          authentication(userName: ossrhUsername, password: ossrhPassword)
+        }
+
+        snapshotRepository(url: "https://oss.sonatype.org/content/repositories/snapshots/") {
+          authentication(userName: ossrhUsername, password: ossrhPassword)
         }
 
         pom.project {
-          name 'Gapi Tools'
+          name 'Google API Tools'
           packaging 'jar'
-          description 'Google API tools frameworks.'
-          url 'http://www.googleapis.com'
+          artifactId 'toolkit'
+          description 'Google API tools framework.'
```

Comment:
Is this the right full description for `toolkit`? I find it somewhat confusing given that the API compiler contains the package `com.google.api.tools.framework` -- I think of toolkit as built on top of the framework, rather than being a framework itself.